zqd/api/bzng.go

```diff
@@ -0,0 +1,84 @@
+package api
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zio/bzngio"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+type BzngSearch struct {
+	reader *bzngio.Reader
+	ctrl   interface{}
+	err    error
+}
+
+func NewBzngSearch(body io.Reader) *BzngSearch {
+	return &BzngSearch{
+		reader: bzngio.NewReader(body, resolver.NewContext()),
+	}
+}
+
+//XXX
+const mtu = 100
+
+func (r *BzngSearch) Pull() (zbuf.Batch, interface{}, error) {
```

Comment:
How would we replace the bare `interface{}` return type here with a more meaningful type?

Reply:
Still working on this... A switch is going to be unavoidable but I'm thinking for wrapping this is in a `SearchCtrl` struct.